soccer/planning/SettlePathPlanner.cpp

```diff
@@ -0,0 +1,528 @@
+            // b is dist down ball line
+            float distDownBallMovementLine = sqrt(stoppingDist*stoppingDist - distToBallMovementLine*distToBallMovementLine);
+            finalStoppingPoint = nearestPointToRobot + distDownBallMovementLine * ballMovementDir;
+        }
+    }
+    
+    // Target stopping point with 0 speed.
+    MotionInstant finalStoppingMotion(finalStoppingPoint, Point(0,0));
+    std:unique_ptr<MotionCommand> directCommand = 
+        std::make_unique<DirectPathTargetCommand>(finalStoppingMotion);
+
+    auto request = PlanRequest(planRequest.systemState,
+                               startInstant,
+                               std::move(directCommand),
+                               planRequest.constraints,
+                               nullptr,
+                               planRequest.obstacles,
+                               planRequest.dynamicObstacles,
+                               planRequest.shellID);
+
+    std::unique_ptr<Path> dampenEnd = directPlanner.run(request);
+    dampenEnd->setDebugText("Damping");
+
+    if (prevPath) {
+        RJ::Time newStartTime = prevPath->startTime();
+        dampenEnd = std::make_unique<CompositePath>(std::move(prevPath),
+                                                    std::move(dampenEnd));
+        dampenEnd->setStartTime(newStartTime);
+    }
+
+    return make_unique<AngleFunctionPath>(
+        std::move(dampenEnd),
+        angleFunctionForCommandType(FacePointCommand(facePos)));
+}
+
+std::unique_ptr<Path> SettlePathPlanner::invalid(const PlanRequest& planRequest) {
+    std::cout << "WARNING: Invalid state in settle planner. Restarting" << std::endl;
+    currentState = Intercept;
+
+    // Stop movement until next frame since it's the safest option programmatically
+    MotionInstant target(planRequest.start);
+    target.vel = Point(0, 0);
+
+    std::unique_ptr<MotionCommand> rrtCommand =
+        std::make_unique<PathTargetCommand>(target);
+
+    auto request = PlanRequest(planRequest.systemState,
+                               planRequest.start,
+                               std::move(rrtCommand),
+                               planRequest.constraints,
+                               nullptr,
+                               planRequest.obstacles,
+                               planRequest.dynamicObstacles,
+                               planRequest.shellID);
+
+    auto path = rrtPlanner.run(request);
+    path->setDebugText("Invalid state in settle");
+
+    return make_unique<AngleFunctionPath>(
+        std::move(path),
+        angleFunctionForCommandType(FacePointCommand(planRequest.systemState.ball.pos)));
+}
+
+void SettlePathPlanner::calcDeltaPosForDir(const Ball& ball, const MotionInstant& startInstant, 
+                                           Geometry2d::Point& deltaRobotPos, Geometry2d::Point& facePos) {
+    // If we have a valid bounce target
+    //
+    // TODO: Simplify the math
+    if (targetBounceDirection) {
```

Comment:
Check my math here. The goal is to offset the position of the robot so that when the robot is facing some other point besides the ball, the ball will hit the mouth of the robot. `deltaRobotPos` is the offset from the ball movement line for the robot settling the ball. `facePos` is the point that we end up trying to face.